query/src/main/java/io/keen/client/java/KeenQueryClient.java

```diff
@@ -0,0 +1,704 @@
+package io.keen.client.java;
+
+import java.io.IOException;
+import java.io.OutputStream;
+import java.io.OutputStreamWriter;
+import java.io.StringReader;
+import java.io.StringWriter;
+import java.net.URL;
+import java.util.ArrayList;
+import java.util.HashMap;
+import java.util.Locale;
+import java.util.Map;
+
+import io.keen.client.java.exceptions.KeenQueryClientException;
+
+import io.keen.client.java.exceptions.ServerException;
+import io.keen.client.java.http.HttpHandler;
+import io.keen.client.java.http.OutputSource;
+import io.keen.client.java.http.Request;
+import io.keen.client.java.http.Response;
+import io.keen.client.java.http.UrlConnectionHttpHandler;
+
+import io.keen.client.java.result.Group;
+import io.keen.client.java.result.QueryResult;
+import io.keen.client.java.result.DoubleResult;
+import io.keen.client.java.result.LongResult;
+import io.keen.client.java.result.StringResult;
+import io.keen.client.java.result.ListResult;
+import io.keen.client.java.result.IntervalResult;
+import io.keen.client.java.result.GroupByResult;
+
+/**
+ * <p>
+ * KeenQueryClient provides all of the functionality required to execute the basic queries
+ * supported by the Data Analysis API: https://keen.io/docs/data-analysis/
+ * </p>
+ * <p> This include Count, Count Unique, Sum, Average, Maxiumum, Minimum, Median,
+ * Percentile, and Select Unique. It does not include Extractions, Multi-Analysis, and Funnels.</p>
+ * */
+public class KeenQueryClient {
+
+    private static final String ENCODING = "UTF-8";
+    private KeenJsonHandler jsonHandler;
```

Comment:
Nit: I believe these properties can all be marked as `final`, since they're set in the constructor and never changed, correct?